[NCCL] Add Error log when ProcessGroupNCCL takes down process upon
timeout/error

osalpekar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#44988 [NCCL] Add Error log when ProcessGroupNCCL takes down process upon
timeout/error**
timeout/error**
timeout/error**

timeout/error

The new NCCL async error handling feature throws an exception from the
workCleanup Thread if one of the NCCL operations encounters an error or times
out. This PR adds an error log to make it more clear to the user why the
training process crashed.

Differential Revision: [D23794801](https://our.internmc.facebook.com/intern/diff/D23794801/)

osalpekar committed:
[NCCL] Add Error log when ProcessGroupNCCL takes down process upon
timeout/error

The new NCCL async error handling feature throws an exception from the
workCleanup Thread if one of the NCCL operations encounters an error or times
out. This PR adds an error log to make it more clear to the user why the
training process crashed.

Differential Revision: [D23794801](https://our.internmc.facebook.com/intern/diff/D23794801/)

[ghstack-poisoned]

pritamdamania87 commented on torch/lib/c10d/ProcessGroupNCCL.cpp:
Can we add a unit test for this case using multiprocessing?

pritamdamania87 reviewed (changes requested):
Thanks for adding this! Requesting changes since I think we need to test this works properly.

osalpekar committed:
Update on "[NCCL] Add Error log when ProcessGroupNCCL takes down process upon
timeout/error"

timeout/error**

timeout/error

The new NCCL async error handling feature throws an exception from the
workCleanup Thread if one of the NCCL operations encounters an error or times
out. This PR adds an error log to make it more clear to the user why the
training process crashed.

Differential Revision: [D23794801](https://our.internmc.facebook.com/intern/diff/D23794801/)

[ghstack-poisoned]

pritamdamania87 reviewed (approved):
Stamping since we've added the test in https://github.com/pytorch/pytorch/pull/46044.

osalpekar committed:
Update on "[NCCL] Add Error log when ProcessGroupNCCL takes down process upon
timeout/error"

timeout/error**
timeout/error**

timeout/error

The new NCCL async error handling feature throws an exception from the
workCleanup Thread if one of the NCCL operations encounters an error or times
out. This PR adds an error log to make it more clear to the user why the
training process crashed.

Differential Revision: [D23794801](https://our.internmc.facebook.com/intern/diff/D23794801/)

[ghstack-poisoned]
